ui/src/app/settings/components/GeneralForm/GeneralForm.js

```diff
@@ -0,0 +1,72 @@
+import { Formik } from "formik";
+import PropTypes from "prop-types";
+import React from "react";
+import * as Yup from "yup";
+
+import Button from "app/base/components/Button";
+import Form from "app/base/components/Form";
+import Input from "app/base/components/Input";
+
+const GeneralSchema = Yup.object().shape({
+  maasName: Yup.string().required(),
+  enableAnalytics: Yup.boolean()
+});
+
+const GeneralForm = ({ maasName, enableAnalytics }) => {
+  return (
+    <Formik
+      initialValues={{
+        maasName,
+        enableAnalytics
+      }}
+      validationSchema={GeneralSchema}
+    >
+      {({
+        errors,
+        isSubmitting,
+        touched,
+        handleSubmit,
+        handleChange,
+        handleBlur,
+        values
+      }) => (
+        <Form onSubmit={handleSubmit}>
+          <Input
+            error={touched.naasName && errors.maasName}
+            id="maasName"
+            label="MAAS name"
+            onBlur={handleBlur}
+            onChange={handleChange}
+            required={true}
+            type="text"
+            value={values.maasName}
+          />
+          <Input
+            error={touched.enableAnalytics && errors.enableAnalytics}
+            id="enableAnalytics"
+            label="Enable Google Analytics in MAAS UI to shape improvements in user experience"
+            onBlur={handleBlur}
+            onChange={handleChange}
+            type="checkbox"
+            value={values.enableAnalytics}
+            checked={values.enableAnalytics}
+          />
+          <Button
+            appearance="positive"
+            type="submit"
+            disabled={isSubmitting || !values.maasName}
+          >
+            Save
```

Comment:
Does this button component provide user feedback via a callback?

Reply:
No, this is a standard vanilla button. I'd like to see the ActionButton we use in rbac moved to the shared library though.